aws-api-appsync/src/main/java/com/amplifyframework/datastore/appsync/SerializedModel.java

```diff
@@ -44,6 +44,45 @@ private SerializedModel(
         this.modelSchema = modelSchema;
     }
 
+    /**
+     * Creates a SerializedModel from a generated Java Model object.
+     * @param model Model object
+     * @param modelSchema schema for the Model object
+     * @param <T> type of the Model object.
+     * @return SerializedModel equivalent of the Model object.
+     */
+    public static <T extends Model> SerializedModel create(T model, ModelSchema modelSchema) {
+        return SerializedModel.builder()
+                .serializedData(ModelConverter.toMap(model))
+                .modelSchema(modelSchema)
+                .build();
+    }
+
+    /**
+     * Computes the difference between two Models, comparing equality of each field value for each Model, and returns
+     * the difference as a SerializedModel.
+     * @param updated the updated Model, whose values will be used to build the resulting SerializedModel.
+     * @param original the original Model to compare against.
+     * @param modelSchema ModelSchema for the Models between compared.
+     * @param <T> type of the Models being compared.
+     * @return a SerializedModel, containing only the values from the updated Model that are different from the
+     *         corresponding values in original.
+     */
+    public static <T extends Model> SerializedModel difference(T updated, T original, ModelSchema modelSchema) {
```

Comment:
Should modelSchema be a parameter or could we just infer it from the models passed in?

Reply:
This would be pretty easy if T was always a generated Java model, but it's more complex since it could be a `SerializedModel` now too.  If we don't pass in a `ModelSchema`, we'd probably need to pass in a `ModelSchemaRegistry`, which doesn't seem much better, so I think it's best to leave as is.